File: workspace/java/src/test/java/com/dslplatform/ocd/values/TestOptArrayDateValue.java
package com.dslplatform.ocd.values;

import com.dslplatform.client.Bootstrap;
import com.dslplatform.ocd.values.OptArrayDateInValue.OptArrayDateValue;
import com.dslplatform.patterns.ServiceLocator;
import org.junit.*;
import static org.junit.Assert.*;

public class TestOptArrayDateValue {
    private static ServiceLocator locator;

    @BeforeClass
    public static void setUpClass() throws Exception {
//        locator = Bootstrap.init(TestOptArrayDateValue.class.getResourceAsStream("dsl-project.ini"));
    }

    @AfterClass
    public static void tearDownClass() throws Exception {
//        locator.resolve(java.util.concurrent.ExecutorService.class).shutdown();
//        locator = null;
    }

    @Before
    public void setUp() throws Exception {
    }

    @After
    public void tearDown() throws Exception {
    }

    @Test
    public void testFieldType() throws NoSuchFieldException {
        assertEquals(
                org.joda.time.LocalDate[].class,
                OptArrayDateValue.class.getDeclaredField("optArrayDate").getGenericType());
    }

    @Test
    public void testGetterType() throws NoSuchMethodException {
        assertEquals(
                org.joda.time.LocalDate[].class,
                OptArrayDateValue.class.getMethod("getOptArrayDate").getGenericReturnType());
    }

    @Test
    public void testSetterType() throws NoSuchMethodException {
        assertEquals(
                OptArrayDateValue.class,
                OptArrayDateValue.class.getMethod("setOptArrayDate", org.joda.time.LocalDate[].class).getReturnType());
    }

    @Test
    public void testDefaultPropertyValue() {
        assertNull(new OptArrayDateValue().getOptArrayDate());
    }
}

File: legacy/workspace/scala/src/test/scala/com/dslplatform/ocd/values/TestOptArrayDateValue.scala
package com.dslplatform.ocd.values

import OptArrayDateInValue.OptArrayDateValue
import com.dslplatform.api.patterns.ServiceLocator
import org.scalatest._
import scala.reflect.runtime.universe._

class TestOptArrayDateValue extends SpecExtensions {
  def `Test property field type: Option[Array[org.joda.time.LocalDate]]`() =
    checkType(
      typeOf[Option[Array[org.joda.time.LocalDate]]]
    , typeOf[OptArrayDateValue].member("optArrayDate": TermName).asMethod.returnType
    )

  def `Test property default value: None`() {
    assert(OptArrayDateValue().optArrayDate === None)
  }
}
